orchestrator/security.py

```diff
@@ -10,41 +10,103 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import importlib
 
 from authlib.integrations.starlette_client import OAuth
 
 from nwastdlib.url import URL
-from oauth2_lib.fastapi import HTTPX_SSL_CONTEXT, OIDCUser, opa_decision, opa_graphql_decision
-from orchestrator.settings import oauth2_settings
+from oauth2_lib.fastapi import (
+    HTTPX_SSL_CONTEXT,
+    AuthenticationFunc,
+    Authorization,
+    GraphqlAuthorization,
+    GraphqlAuthorizationFunc,
+    OIDCAuth,
+)
+from oauth2_lib.settings import oauth2lib_settings
+from orchestrator.settings import auth_settings
 
 oauth_client_credentials = OAuth()
 
-well_known_endpoint = URL(oauth2_settings.OIDC_CONF_WELL_KNOWN_URL)
-
 oauth_client_credentials.register(
     "connext",
-    server_metadata_url=well_known_endpoint / ".well-known" / "openid-configuration",
-    client_id=oauth2_settings.OAUTH2_RESOURCE_SERVER_ID,
-    client_secret=oauth2_settings.OAUTH2_RESOURCE_SERVER_SECRET,
+    server_metadata_url=URL(oauth2lib_settings.OIDC_CONF_URL),
+    client_id=oauth2lib_settings.OAUTH2_RESOURCE_SERVER_ID,
+    client_secret=oauth2lib_settings.OAUTH2_RESOURCE_SERVER_SECRET,
     request_token_params={"grant_type": "client_credentials"},
     client_kwargs={"verify": HTTPX_SSL_CONTEXT},
```

Comment:
It seems we're not using this `oauth_client_credentials` object or its return value anywhere, also checked in our surf orchestrator. If it's the same for you then I think we can remove this?

Or perhaps move it to oauth2-lib and use it for interacting with the OIDC backend. It has some built-in logic for verifying tokens https://docs.authlib.org/en/latest/client/starlette.html#register-remote-apps I'm not sure how to use it though.

Reply:
We also don't use it in our code and I wasn't sure if it causes bug if I remove it so that I decided to leave it and adapt it with new changes 

> It seems we're not using this `oauth_client_credentials` object or its return value anywhere, also checked in our surf orchestrator. If it's the same for you then I think we can remove this?
> 
> Or perhaps move it to oauth2-lib and use it for interacting with the OIDC backend. It has some built-in logic for verifying tokens https://docs.authlib.org/en/latest/client/starlette.html#register-remote-apps I'm not sure how to use it though.